library/core/src/num/f32.rs

```diff
@@ -1054,6 +1059,8 @@ impl f32 {
                   without modifying the original"]
     #[stable(feature = "float_approx_unchecked_to", since = "1.44.0")]
     #[inline]
+    // is_finite() checks if the given float is neither infinite nor NaN.
+    #[requires(self.is_finite() && kani::float::float_to_int_in_range::<Self, Int>(self))]
```

Comment:
Can you please add this function to `ub_checks::` module so the contract is not Kani specific?

You would need to add a dummy method here:
https://github.com/model-checking/verify-rust-std/blob/7e8a03d361af17f1db3d3fe7be9501589fd5a01a/library/core/src/ub_checks.rs#L181
and exported it here:
https://github.com/model-checking/verify-rust-std/blob/7e8a03d361af17f1db3d3fe7be9501589fd5a01a/library/core/src/ub_checks.rs#L226

Reply:
@celinval Do you mean like:
```rust
/// Check if a float is representable in the given integer type
pub fn float_to_int_in_range<Float, Int>(value: Float) -> bool
where
    Float: FloatToInt<Int> { // Will need to import `core::convert::FloatToInt`
    let _ = value;
    true
}
```

Reply:
And then change the contract to:
```rust
#[requires(self.is_finite() && ub_checks::float_to_int_in_range::<Self, Int>(self))]
```
?